pandas/core/reshape/merge.py

```diff
@@ -1028,7 +1028,14 @@ def _validate_specification(self):
                 common_cols = self.left.columns.intersection(
                     self.right.columns)
                 if len(common_cols) == 0:
-                    raise MergeError('No common columns to perform merge on')
+                    raise MergeError('No common columns to perform merge on. '
+                                     'Merge options: right_on={ron}, '
+                                     'left_on={lon}, right_index={ridx}, '
+                                     'left_index={lidx}'
+                                     .format(ron=self.right_on,
+                                             lon=self.left_on,
```

Comment:
can you update the test for this (it prob just raises) and match the error message with ``assert_raises_regex``